gems/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb

```diff
@@ -28,6 +28,16 @@ class AssumeRoleCredentials
     # @option options [Integer] :duration_seconds
     # @option options [String] :external_id
     # @option options [STS::Client] :client
+    # @option options [Callable] before_refresh Proc called before
```

Comment:
Does it make sense to push this to refreshable credentials and have all refreshable credentials support this?

Reply:
Yeah - I was trying to think through that.  I think the key part is the documentation, which likely should be per class.... and to be useful, each credential provider may need to provide other accessors.  But I think the actual call to before_refresh could be done in the RefreshingCredentials `refresh!` method.